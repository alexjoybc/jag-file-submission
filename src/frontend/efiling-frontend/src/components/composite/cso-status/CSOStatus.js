import React from "react";
import PropTypes from "prop-types";
import ConfirmationPopup from "shared-components";
import { propTypes } from "../../../types/propTypes";

export default function CSOStatus({
  csoStatus: { accountExists, confirmationPopup }
}) {
  return (
    <div>
      {accountExists && <p>Account exists! Proceed</p>}
      {!accountExists && <p>Account does not exist, form will be here</p>}

      <ConfirmationPopup
        modal={confirmationPopup.modal}
        mainButton={confirmationPopup.mainButton}
        confirmButton={confirmationPopup.confirmButton}
        cancelButton={confirmationPopup.cancelButton}
      />
    </div>
  );
}

CSOStatus.propTypes = {
  csoStatus: PropTypes.shape({
    accountExists: PropTypes.bool.isRequired,
    confirmationPopup: propTypes.confirmationPopup
  }).isRequired
};
