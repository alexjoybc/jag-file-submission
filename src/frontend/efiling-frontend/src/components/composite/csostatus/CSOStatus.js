import React from "react";
import PropTypes from "prop-types";
import ConfirmationPopup from "shared-components";
import { propTypes } from "../../../types/propTypes";

export default function CSOStatus({ accountExists, confirmationPopup }) {
  return (
    <div>
      {accountExists && <p>Account exists! Proceed</p>}
      {!accountExists && <p>Account does not exist, form will be here</p>}

      <ConfirmationPopup
        modal={confirmationPopup.modal}
        mainButton={confirmationPopup.mainButton}
        confirmButton={confirmationPopup.confirmButton}
        cancelButton={confirmationPopup.cancelButton}
      />
    </div>
  );
}

CSOStatus.propTypes = {
  accountExists: PropTypes.bool.isRequired,
  confirmationPopup: propTypes.confirmationPopup.isRequired
};
