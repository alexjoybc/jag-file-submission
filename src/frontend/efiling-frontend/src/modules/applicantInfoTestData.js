const applicantInfo = {
  bceID: "bobross42",
  firstName: "Bob",
  lastName: "Ross",
  emailAddress: "bob.ross@example.com"
};

export function getApplicantInfo() {
  return applicantInfo;
}
