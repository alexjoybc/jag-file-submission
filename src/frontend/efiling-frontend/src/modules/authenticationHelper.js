const jwt = require("jsonwebtoken");

export function getJWTData() {
  const token = localStorage.getItem("jwt");
  if (!token) return false;

  // get the decoded payload and header
  const decoded = jwt.decode(token, { complete: true });
  return decoded.payload;
}

export function generateJWTToken(payload) {
  return jwt.sign(payload, "something");
}
