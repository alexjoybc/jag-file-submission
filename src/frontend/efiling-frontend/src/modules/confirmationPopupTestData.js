/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn"
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width"
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width"
};

const handleClose = () => {};
const handleShow = () => {};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: true,
    handleShow,
    handleClose,
    handleConfirm: () => {},
    title: "title",
    body
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
