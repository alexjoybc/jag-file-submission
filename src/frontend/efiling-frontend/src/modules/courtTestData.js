const court = {
  locationDescription: "Court location",
  fileNumber: "Court file number",
  levelDescription: "Level",
  classDescription: "Class",
  level: "P",
  courtClass: "F",
};

export function getCourtData() {
  return court;
}
