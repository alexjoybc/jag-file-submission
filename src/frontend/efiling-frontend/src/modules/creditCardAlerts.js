import React from "react";
import { MdCreditCard } from "react-icons/md";
import { Alert } from "shared-components";

const existingCreditCard = () => {
  return (
    <Alert
      icon={<MdCreditCard size={32} />}
      type="success"
      styling="success-background no-padding-bottom"
      element={
        <p>
          <span>
            You have a valid Credit Card registered with your CSO account.
          </span>
          <br />
          <a
            href="https://justice.gov.bc.ca/cso/about/index.do"
            target="_blank"
            rel="noopener noreferrer"
          >
            Register a new Credit Card.
          </a>
        </p>
      }
    />
  );
};

const noCreditCard = () => {
  return (
    <Alert
      icon={<MdCreditCard size={32} />}
      type="error"
      styling="error-background no-padding-bottom"
      element={
        <p>
          <span>
            You do not have a valid Credit Card registered with your CSO
            account.
          </span>
          <br />
          <a
            href="https://justice.gov.bc.ca/cso/about/index.do"
            target="_blank"
            rel="noopener noreferrer"
          >
            Register a Credit Card now
          </a>
          &nbsp;to continue.
        </p>
      }
    />
  );
};

export function getCreditCardAlerts() {
  const existingCreditCardData = existingCreditCard();
  const noCreditCardData = noCreditCard();

  return {
    existingCreditCard: existingCreditCardData,
    noCreditCard: noCreditCardData,
  };
}
