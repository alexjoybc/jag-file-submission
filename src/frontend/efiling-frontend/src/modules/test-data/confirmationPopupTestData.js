/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const onButtonClick = () => {};

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn",
  onClick: onButtonClick,
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width",
  onClick: onButtonClick,
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width",
  onClick: onButtonClick,
};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: false,
    title: "title",
    body,
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
