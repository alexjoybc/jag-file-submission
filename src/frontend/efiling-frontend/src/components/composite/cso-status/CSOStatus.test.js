import React from "react";
import { render } from "@testing-library/react";
import { getTestData } from "../../../modules/confirmationPopupTestData";

import CSOStatus from "./CSOStatus";

describe("CSOStatus Component", () => {
  const confirmationPopup = getTestData();

  test("Matches the snapshot when account exists", () => {
    const { asFragment } = render(
      <CSOStatus csoStatus={{ accountExists: true, confirmationPopup }} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Matches the snapshot when account does not exist", () => {
    const { asFragment } = render(
      <CSOStatus csoStatus={{ accountExists: false, confirmationPopup }} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
