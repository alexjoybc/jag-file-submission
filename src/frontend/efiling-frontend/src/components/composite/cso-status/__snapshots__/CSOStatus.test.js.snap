// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSOStatus Component Matches the snapshot when account does not exist 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="content col-md-8"
    >
      <p>
        Account does not exist, form will be here
      </p>
      <div>
        <button
          class="normal-white btn"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CSOStatus Component Matches the snapshot when account exists 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="content col-md-8"
    >
      <p>
        Account exists! Proceed
      </p>
      <div>
        <button
          class="normal-white btn"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
